src/hooks/useChainMetadata.ts

```diff
@@ -0,0 +1,31 @@
+import { toRefs, reactive, watchEffect } from 'vue';
+import { useApi } from '.';
+import { setDefaultUnitName } from '@/helper/units';
+
+interface Metadata {
+  decimal: number;
+  defaultUnitToken: string;
+}
+
+export const useChainMetadata = () => {
+  const { api } = useApi();
+
+  const state = reactive<Metadata>({
+    decimal: 12,
+    defaultUnitToken: '',
+  });
+
+  watchEffect(() => {
+    const registry = api?.value?.registry;
+
+    const decimals = registry?.chainDecimals;
+    const tokens = registry?.chainTokens;
+
+    state.decimal = (decimals || [])[0];
+    state.defaultUnitToken = (tokens || [])[0];
+
+    setDefaultUnitName(state.defaultUnitToken);
```

Comment:
This part is showing some strange behavior for Shiden network.
Astar and Dusty correctly displays `PLM` and `PLD` but when connected to Shiden, it shows `Unit SDN`
![bandicam 2021-07-09 15-41-38-798](https://user-images.githubusercontent.com/40356749/125073437-00b96680-e0cd-11eb-8915-b6e7d9a3e277.jpg)

Reply:
![bandicam 2021-07-09 15-55-53-633](https://user-images.githubusercontent.com/40356749/125074461-5cd0ba80-e0ce-11eb-8952-d7cb9bb7f5ab.jpg)
FYI accessing Shiden network endpoint without resolving [this](https://github.com/PlasmNetwork/plasm-apps-vue/pull/43#discussion_r666884661) issue will result in a node type registry error and only render 0 SDN for any accounts.
Error log:

```
logger.js?eb80:47 2021-07-09 15:55:20        RPC-CORE: getStorage(key: StorageKey, at?: BlockHash): StorageData:: Unable to decode storage system.account:: createType(AccountInfo):: {"nonce":"Index","consumers":"RefCount","providers":"RefCount","data":"AccountData"}:: Decoded input doesn't match input, received 0x0000000000000000010000000000000000001b9ce6fdddaf3500000000000000…0000000000000000000000000000000000000000000000000000000000000000 (80 bytes), created 0x0000000000000000010000000000000000001b9ce6fdddaf3500000000000000…0000000000000000000000000000000000000000000000000000000000000000 (76 bytes)
apply @ logger.js?eb80:47
index.js?33d6:433 Uncaught (in promise) Error: Unable to decode storage system.account:: createType(AccountInfo):: {"nonce":"Index","consumers":"RefCount","providers":"RefCount","data":"AccountData"}:: Decoded input doesn't match input, received 0x0000000000000000010000000000000000001b9ce6fdddaf3500000000000000…0000000000000000000000000000000000000000000000000000000000000000 (80 bytes), created 0x0000000000000000010000000000000000001b9ce6fdddaf3500000000000000…0000000000000000000000000000000000000000000000000000000000000000 (76 bytes)
    at RpcCore._newType (webpack-internal:///./node_modules/@polkadot/rpc-core/index.js:449)
    at RpcCore._formatStorageData (webpack-internal:///./node_modules/@polkadot/rpc-core/index.js:383)
    at RpcCore._formatOutput (webpack-internal:///./node_modules/@polkadot/rpc-core/index.js:360)
    at callWithRegistry (webpack-internal:///./node_modules/@polkadot/rpc-core/index.js:239)
```

you can make this function
https://github.com/PlasmNetwork/plasm-apps-vue/blob/1209def72e639ffa03174ae21dbd52771e39cebb/src/api/polkadot/connectApi.ts#L58
pass these type values to see the balance as a quick patch (copied from [`plasmCollatorDefinitions`](https://github.com/PlasmNetwork/api/blob/2d52d49d8e95dd067c289578593ab340fbcda4b4/packages/types/src/networkSpecs/plasmCollator.ts#L3))

```ts
  const api = new ApiPromise({
    provider,
    types: {
      Address: 'AccountId',
      LookupSource: 'AccountId',
      ChainId: {
        _enum: {
          RelayChain: null,
          Parachain: 'ParaId',
        },
      },
      XCurrencyId: {
        chain_id: 'ChainId',
        currency_id: 'Bytes',
      },
      CurrencyIdOf: 'CurrencyId',
      CurrencyId: {
        _enum: {
          Token: 'TokenSymbol',
        },
      },
      TokenSymbol: {
        _enum: ['ACA', 'AUSD', 'DOT', 'XBTC', 'LDOT', 'RENBTC', 'SDN', 'PLM'],
      },
      AmountOf: 'Amount',
      Amount: 'i128',
    },
  });
```

Reply:
it seems resolved after fixing above

Reply:
thank you!